Mathlib/Dynamics/PeriodicPts/Lemmas.lean

```diff
@@ -43,6 +46,12 @@ theorem minimalPeriod_eq_prime_iff {p : ℕ} [hp : Fact p.Prime] :
     iff_self_and]
   exact fun h ↦ ne_of_eq_of_ne h hp.out.ne_one
 
+theorem minimalPeriod_eq_sInf_n_pos_IsPeriodicPt :
+    -- alternative set expression: `{ n | 0 < n ∧ IsPeriodicPt f n x }`
+    minimalPeriod f x = sInf { n > 0 | IsPeriodicPt f n x } := by
+  dsimp [minimalPeriod, periodicPts, sInf]
+  grind -- don't know why `rfl` doesn't work here
+
```

Comment:
The two sides of the equation use different `DecidablePred` instances for `Nat.find`.